File: dotnet-elasticemail-examples/Examples/WithAttachments.cs
namespace ElasticEmailExamples;

using System.Text;
using ElasticEmail.Api;
using ElasticEmail.Client;
using ElasticEmail.Model;

public static class WithAttachments
{
    public static async Task RunAsync()
    {
        var emailsApi = new EmailsApi(Ee.Config());

        var fileContent = "Sample Attachment\n==================\n\nThis file was attached to your email.\n"
            + $"Sent at: {DateTime.UtcNow:O}\n";

        try
        {
            var result = await emailsApi.EmailsTransactionalPostAsync(new EmailTransactionalMessageData(
                recipients: new TransactionalRecipient(to: new List<string> { Ee.To }),
                content: new EmailContent(
                    from: Ee.From,
                    subject: "Email with Attachment",
                    body: new List<BodyPart>
                    {
                        new BodyPart(BodyContentType.HTML,
                            "<h1>Your attachment is ready</h1><p>Please find the file attached to this email.</p>"),
                    },
                    // BinaryContent is sent as base64. Total message size limit applies (see account limits).
                    attachments: new List<MessageAttachment>
                    {
                        new MessageAttachment(
                            binaryContent: Encoding.UTF8.GetBytes(fileContent),
                            name: "sample.txt",
                            contentType: "text/plain"),
                    })));

            Console.WriteLine("Email with attachment sent successfully!");
            Console.WriteLine($"Transaction ID: {result.TransactionID}");
            Console.WriteLine($"Message ID: {result.MessageID}");
        }
        catch (ApiException e)
        {
            Ee.Fail("send email", e);
        }
    }
}

File: rust-elasticemail-examples/examples/with_attachments.rs
use rust_elasticemail_examples::{config, from, print_api_error, to};
use ElasticEmail::apis::emails_api;
use ElasticEmail::models::{
    BodyContentType, BodyPart, EmailContent, EmailTransactionalMessageData, MessageAttachment, TransactionalRecipient,
};

#[tokio::main]
async fn main() {
    let config = config();

    let file_content = format!(
        "Sample Attachment\n==================\n\nThis file was attached to your email.\nSent at: {}\n",
        chrono::Utc::now().to_rfc3339()
    );

    // The SDK base64-encodes BinaryContent when serializing, so pass raw bytes.
    // Total message size limit applies (see account limits).
    let attachment = MessageAttachment {
        content_type: Some("text/plain".to_string()),
        ..MessageAttachment::new(file_content.into_bytes(), "sample.txt".to_string())
    };

    let content = EmailContent {
        subject: Some("Email with Attachment".to_string()),
        body: Some(vec![BodyPart {
            content: Some("<h1>Your attachment is ready</h1><p>Please find the file attached to this email.</p>".to_string()),
            ..BodyPart::new(BodyContentType::Html)
        }]),
        attachments: Some(vec![attachment]),
        ..EmailContent::new(from())
    };

    let data = EmailTransactionalMessageData::new(TransactionalRecipient::new(vec![to()]), content);

    match emails_api::emails_transactional_post(&config, data).await {
        Ok(result) => {
            println!("Email with attachment sent successfully!");
            println!("Transaction ID: {}", result.transaction_id.unwrap_or_default());
            println!("Message ID: {}", result.message_id.unwrap_or_default());
        }
        Err(e) => {
            print_api_error("sending email", &e);
            std::process::exit(1);
        }
    }
}
